CMakeLists.txt

```diff
@@ -89,13 +89,69 @@ if(STABLEHLO_EXTERNAL_PROJECT_BUILD)
   list(APPEND CMAKE_MODULE_PATH "${MLIR_MAIN_SRC_DIR}/cmake/modules")
 elseif(NOT STABLEHLO_BUILD_EMBEDDED)
   message(STATUS "Building StableHLO with an installed MLIR")
-  find_package(MLIR REQUIRED CONFIG)
+
+  # These defaults are moderately important to us, but the user *can*
+  # override them (enabling some of these brings in deps that will conflict,
+  # so ymmv).
+  # https://github.com/openxla/iree/blob/f3b6bcd79b24ef4a9b355eb3f3496ffafcbd0881/build_tools/cmake/iree_llvm.cmake#L127
+  set(LLVM_INCLUDE_EXAMPLES OFF CACHE BOOL "")
+  set(LLVM_INCLUDE_TESTS OFF CACHE BOOL "")
+  set(LLVM_INCLUDE_BENCHMARKS OFF CACHE BOOL "")
+  set(LLVM_APPEND_VC_REV OFF CACHE BOOL "")
+  set(LLVM_ENABLE_IDE ON CACHE BOOL "")
+  set(LLVM_ENABLE_BINDINGS OFF CACHE BOOL "")
+  # LLVM defaults to building all targets. We always enable targets that we need
+  # as we need them, so default to none. The user can override this as needed,
+  # which is fine.
+  set(LLVM_TARGETS_TO_BUILD "" CACHE STRING "")
+
+  # We enable LLVM projects as needed. The user can override this.
+  set(LLVM_ENABLE_PROJECTS "" CACHE STRING "")
+  set(LLVM_EXTERNAL_PROJECTS "" CACHE STRING "")
+
+  # Unconditionally enable mlir.
+  list(APPEND LLVM_ENABLE_PROJECTS mlir)
+
+  # Setup LLVM lib and bin directories.
+  set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/llvm-project/bin)
+  set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/llvm-project/lib)
+  set(LLVM_TOOLS_BINARY_DIR ${CMAKE_BINARY_DIR}/llvm-project/bin)
```

Comment:
having to set these here are pretty ugly; we can discuss if this is an adequate trade-off.